lib/philomena/images/query.ex

```diff
@@ -12,6 +14,39 @@ defmodule Philomena.Images.Query do
     end
   end
 
+  defp filter_id_transform(%{filter: true}, _value),
+    do: {:error, "Filter queries inside filters are not allowed."}
+
+  defp filter_id_transform(%{user: user} = ctx, value) do
+    case Integer.parse(value) do
+      {value, ""} when value >= 0 ->
+        filter =
+          case ctx do
```

Comment:
with {:ok, filter} <- Map.fetch(ctx.filters, value),
         true <- Canada.Can.can?(user, :show, filter),
  ... do
else
  e when e in [:error, false] ->
  {:error, "Invalid filter ..."} 
end

Reply:
Could probably also add the integer.parse into it